client/javascript-esm/packages/jolokia/src/jolokia-utils.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  JolokiaErrorResponse,
+  JolokiaSuccessResponse,
+  VersionResponseValue
+} from "./jolokia-types.js"
+
+export type ParseResult<T> = { hasError: false; parsed: T } | { hasError: true; error: string }
+
+function isObject(value: unknown): value is object {
+  const type = typeof value
+  return value != null && (type === 'object' || type === 'function')
+}
+
+export function isResponseSuccessType(o: unknown): o is JolokiaSuccessResponse {
+  return isObject(o) && 'status' in o && 'timestamp' in o && 'value' in o
+}
+
+export function isResponseErrorType(o: unknown): o is JolokiaErrorResponse {
+  return isObject(o) && 'error_type' in o && 'error' in o
+}
+
+export function isVersionResponseType(o: unknown): o is VersionResponseValue {
+  return isObject(o) && 'protocol' in o && 'agent' in o
+}
+
+/**
+ * Parses a jolokia response body and determine its type.
+ *
+ * If the body is an error then the ParseResult
+ * - hasError flag is set to true
+ * - error property is populated
+ *
+ * If the body is successfully parsed then the ParseResult
+ * - hasError flag is set to false
+ * - parsed property is populated
+ */
+export async function responseParse(response: unknown): Promise<ParseResult<JolokiaSuccessResponse | JolokiaErrorResponse>> {
```

Comment:
> OTOH, it's not clear why we need `responseParse` in jolokia.js.

Now I read the function again. I understand the idea, but firstly we no longer seem to need `async` here, as you removed `response.json()` from the original function. Next, `try`-`catch` is also no longer needed as the code now shouldnt throw any exception.

And after being aware that `response.json()` is removed, the function now just checks whether the given unknown object is success or error. At this point, I don't see there's any generic value in the function. For most Jolokia client apps, we can just repeat the same if-else within the app code to achieve the same thing.

Reply:
Yeah. Agree with that.